### DISCUSSION TITLE
New Install 2.3.70 Fails - dockerregisry Boot loop

### DISCUSSION THREAD

**Discussion by xaqpuyrthnbg** [Security-Onion-Solutions/securityonion#5700] [Unsupported Versions] [answered]
Docker Registry is in a start/stop loop; I have 64 GB of RAM; 300 GB of Storage; 2 CPUs.

Everything else seems to be working fine.  

I have tried docker refresh, reboots, running setup again, etc.

I cannot seem to find a specific error for this container.  The /root/errors.log is empty for some reason. 

Update:  I ran another state.apply and found these errors: https://pastebin.com/ainkjbj3

docker-refresh has this output, could this be an issue:

Get http://securityonionv1:5000/v2/: dial tcp 127.0.0.1:5000: connect: connection refused

securityonionv1 log]$ tail -f docker-registry/
tail: error reading ‘docker-registry/’: Is a directory

Thanks,

**Comment by m0duspwnens** [comment, marked as answer]
@thegodsquirrel Looking at the logs, it appears that `/etc/pki/registry.key` and `/etc/pki/registry.crt` were created as directories as opposed to files. This can happen if the Docker container starts prior to Salt creating the files. On your manager/standalone node, we can delete those directories then rerun Salt to create the key and cert. 

On the manager/standalone node:
`rm -rf /etc/pki/registry.key /etc/pki/registry.crt`

It may be worth checking the `/etc/pki/` directory and verify if there are any other `.key` or `.crt` files that were created as directories and removing them.

After that is done, run:
`salt-call state.highstate -l info`

**Comment by xaqpuyrthnbg** [comment]
Thank you for responding!  I was definitely not working on the right problem.  
I ran the recommended commands and then double checked for .crt .key directories and found none.  
After running the highstate it still creates it as a directory:

[INFO    ] Running state [/etc/pki/registry.key] at time 15:17:44.524527
[INFO    ] Executing state x509.private_key_managed for [/etc/pki/registry.key]
............................++
................................................................................................................++
[ERROR   ] Specified target /etc/pki/registry.key is a directory
[INFO    ] Completed state [/etc/pki/registry.key] at time 15:17:46.052481 (duration_in_ms=1527.954)
[INFO    ] State result does not match retry until value, state will be re-run in 30 seconds

**Reply by xaqpuyrthnbg** [reply]
Follow-up:  I touched the files manually and then ran `salt-call state.highstate -l info` twice and now everything is status "OK" now.

**Reply by m0duspwnens** [reply]
Thanks for the update. It appears that `so-dockerregistry' may have been trying to restart and created those as directories even after they were removed. By touching them as files, it preventing them from getting created as directories and Salt was able to update the file with the proper contents.

**Reply by xaqpuyrthnbg** [reply]
Awesome.  Thank you for the help!
